conwayste/src/ui/pane.rs

```diff
@@ -145,32 +133,16 @@ impl Widget for Pane {
         Ok(())
     }
 
-    fn translate(&mut self, dest: Vector2<f32>)
-    {
+    fn translate(&mut self, dest: Vector2<f32>) {
         self.dimensions.translate(dest);
     }
 
     fn on_hover(&mut self, point: &Point2<f32>) {
         self.hover = within_widget(point, &self.dimensions);
-        for w in self.widgets.iter_mut() {
-            w.on_hover(point);
-        }
     }
 
-    fn on_click(&mut self, point: &Point2<f32>) -> Option<(WidgetID, UIAction)> {
-        let hover = self.hover;
-        self.hover = false;
-
-        if hover {
-            for w in self.widgets.iter_mut() {
-                if within_widget(point, &w.rect()) {
-                    let ui_action = w.on_click(point);
-                    if ui_action.is_some() {
-                        return ui_action;
-                    }
-                }
-            }
-        }
+    fn on_click(&mut self, _point: &Point2<f32>) -> Option<(WidgetID, UIAction)> {
+        // TODO: Need to pass children nodes
```

Comment:
I saw you removed the widgets Vec. Is the idea here that the Layering owns the widgets inside a Pane and passes them into Pane as needed? In my UIContext branch I was assuming Panes would own their widgets still.

Reply:
LOL, I just read this in the `id_tree` crate description:
> In this implementation, the Tree owns all of the Nodes and all inter-Node relationships are managed with NodeIds.

I guess that answers that :grinning: